### PR TITLE
Added optional message argument in throwIf (#5772)

### PR DESCRIPTION
**Pull request by vdimir**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Improvement

New optional argument added to `throwIf`. It must be constant string and shown instead of standard exception message

### CODE REVIEW

File: dbms/src/Functions/throwIf.cpp [hunk 1]
```diff
@@ -32,46 +35,68 @@ class FunctionThrowIf : public IFunction
         return name;
     }
 
+    bool isVariadic() const override { return true; }
     size_t getNumberOfArguments() const override
     {
         return 1;
```
Comment (alexey-milovidov) [inline dbms/src/Functions/throwIf.cpp]: Please change to zero.

File: dbms/src/Functions/throwIf.cpp [hunk 2]
```diff
@@ -32,46 +35,68 @@ class FunctionThrowIf : public IFunction
...
 
+        if (number_of_arguments > 1 && !isString(arguments[1]))
+            throw Exception{"Illegal type " + arguments[1]->getName() + " of argument of function " + getName(),
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+
         return std::make_shared<DataTypeUInt8>();
     }
 
     bool useDefaultImplementationForConstants() const override { return true; }
```
Comment (alexey-milovidov) [inline dbms/src/Functions/throwIf.cpp]: Add getArgumentsThatAreAlwaysConstant returning {1}

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Ok. Only a few changes remain.

**Comment by Vdimir** [comment]
I made one excess commit and decide to squash it to one to keep history clean. But changes have been already pushed and it requires force-push. I hope it is not a big problem.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**